backend/app/services/db/rating_manager.py

```diff
@@ -0,0 +1,187 @@
+from datetime import datetime
+
+from .db_connect import connect
+from .user_manager import get_user
+
+
+def rate_profile(rater_name: str, rated_name: str, rating: int):
+    """Rates an existing profile.
+
+    Args:
+        rater_name (str): The username of the rating user.
+        rated_name (str): The username of the user receiving a rating.
+        rating (int): The rating received by the receiving user. Ranges from 1 to 5.
+
+    Returns:
+        True if the function succeeds, else False.
+    """
+
+    with connect() as conn:
+        cursor = conn.cursor()
+
+        # Ensure users cannot rate themselves
+        if rater_name == rated_name:
+            return False
+
+        # Check if users exist
+        userA_details = get_user(rater_name, is_username=True)
+        userB_details = get_user(rated_name, is_username=True)
+
+        if not userA_details or not userB_details:
+            return False
+
+        # Check if a rating already exists
+        cursor.execute(f'''
+                       SELECT * FROM ratings
+                       WHERE rater_name = \"{rater_name}\" AND rated_name = \"{rated_name}\"
```

Comment:
Vulnerable to SQL injection. Use query parameterization for queries from the backend.